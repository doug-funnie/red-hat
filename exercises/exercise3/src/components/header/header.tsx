// tslint:disable:max-line-length
import glamorous from 'glamorous';
import * as React from 'react';
import { white } from 'styles/colors';
import { ThemedProps } from 'styles/theme';

export const height = 60;

export interface Props {
  title?: string;
  onBackButtonPress?(): void;
}

const defaultProps: Partial<Props> = {
  title: 'Red Hat User App',
  onBackButtonPress: undefined,
};

export const HeaderElement = glamorous.header<ThemedProps>(({ theme }) => ({
  alignItems: 'center',
  backgroundColor: theme.colors.red,
  display: 'flex',
  height,
  left: 0,
  padding: '0 15px',
  position: 'fixed',
  right: 0,
  top: 0,
}));

export const IconButton = glamorous.div<ThemedProps<{ onClick?(): void }>>({
  alignItems: 'center',
  borderRadius: '50%',
  cursor: 'pointer',
  display: 'flex',
  height: 35,
  justifyContent: 'center',
  marginRight: 10,
  width: 35,
  ':hover': {
    backgroundColor: 'rgba(0,0,0,0.2)',
  },
});

export const Title = glamorous.div<ThemedProps>(({ theme }) => ({
  color: theme.colors.white,
  fontSize: 18,
}));

export const Header: React.SFC<Props> = ({ title, onBackButtonPress }) => {
  return (
    <HeaderElement>
      {!!onBackButtonPress && (
        <IconButton onClick={onBackButtonPress} role='button'>
          <svg width={20} height={20} viewBox='0 0 24.00 24.00'>
            <path
              fill={white}
              fillOpacity='1'
              strokeWidth='0.2'
              strokeLinejoin='round'
              d='M 20,11L 20,13L 7.98958,13L 13.4948,18.5052L 12.0806,19.9194L 4.16116,12L 12.0806,4.08058L 13.4948,5.49479L 7.98958,11L 20,11 Z '
            />
          </svg>
        </IconButton>
      )}
      <Title>{title}</Title>
    </HeaderElement>
  );
};

Header.defaultProps = defaultProps;
