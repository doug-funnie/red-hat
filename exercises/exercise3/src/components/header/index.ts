export { Header, height } from './header';
