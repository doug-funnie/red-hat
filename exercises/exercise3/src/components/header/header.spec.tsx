import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { Header, IconButton, Props, Title } from './header';

const props: Props = {
  title: '',
  onBackButtonPress: jest.fn(),
};

it('calls onBackButtonPress', () => {
  const view = shallow(<Header {...props} />);
  view.find(IconButton).simulate('click');
  expect(props.onBackButtonPress).toBeCalled();
});

it('does not render IconButton if onBackButtonPress is not provided', () => {
  const view = shallow(<Header {...props} onBackButtonPress={undefined} />);
  expect(view.find(IconButton).length).toBe(0);
});

it('defaults title', () => {
  const view = shallow(<Header {...props} title={undefined} />);
  expect(toJson(view.find(Title))).toMatchSnapshot();
});
