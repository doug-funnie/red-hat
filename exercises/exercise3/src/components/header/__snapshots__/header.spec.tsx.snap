// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaults title 1`] = `
<glamorous(div)>
  Red Hat User App
</glamorous(div)>
`;
